mne/utils/progressbar.py

```diff
@@ -5,99 +5,72 @@
 # License: BSD (3-clause)
 
 from collections.abc import Iterable
-import time
-import logging
-import tempfile
-import sys
-import os.path as op
 import os
-import shutil
+import os.path as op
+import tempfile
+from threading import RLock
 
 import numpy as np
 
-from ._logging import logger
-from .misc import sizeof_fmt
+
+def _noop():
+    pass
 
 
 class ProgressBar(object):
     """Generate a command-line progressbar.
 
     Parameters
     ----------
-    max_value : int | iterable
-        Maximum value of process (e.g. number of samples to process, bytes to
-        download, etc.). If an iterable is given, then `max_value` will be set
-        to the length of this iterable.
+    iterable : iterable | None
+        ``max_value` will be set to the length of this iterable.
     initial_value : int
         Initial value of process, useful when resuming process from a specific
         value, defaults to 0.
     mesg : str
         Message to include at end of progress bar.
-    max_chars : int | str
-        Number of characters to use for progress bar itself.
-        This does not include characters used for the message or percent
-        complete. Can be "auto" (default) to try to set a sane value based
-        on the terminal width.
-    progress_character : char
-        Character in the progress bar that indicates the portion completed.
-    spinner : bool
-        Show a spinner.  Useful for long-running processes that may not
-        increment the progress bar very often.  This provides the user with
-        feedback that the progress has not stalled.
```

Comment:
do we loose this benefit with `tqdm` ?

Reply:
It doesn't do a spinner. It does continuous time estimates, which are better and still update continuously.